### DISCUSSION TITLE
How to test if a form submits altered data?

### DISCUSSION THREAD

**Discussion by alexWaligorski** [neuefische/questions#562] [Web] [answered]
Hi everyone!

It's testing time again. The most frustrating moment before posting a PR...
Anyways, I am trying to test the "edit" feature of the form in my project.
I am trying this by checking if the form which is filled with default data submits the altered entries, when data is changed/edited by the user.
You can find my code [here](https://github.com/alexWaligorski/capstone-project/pull/12/files#diff-aa6074203ce1a30c52a2a1d4e68db2928fc50be670c77a520ece1109eddd7b4b) .

It's the NewMeetingForm component + test.

When i run the test, the error says the received object after the onSubmit function is called is a SyntheticBaseEvent which only includes the default data.
I tried to google similar errors and also went through the github testlogs...which helped fixing some bugs but not the Event issue :(

Please help me checking, if the altered Data is submitted corretly.
Thanks a lot in advance!

**Comment by djfarly** [comment, marked as answer]
Hej Alexandra,

First of all: Awesome that you're writing tests.

Your NewMeetingForm component is simply passing the `onSubmit` prop to the form component. That is why it is called with an submit event. React does some magic with events which is why it is called a `SyntheticBaseEvent`.

You could change your code [here](https://github.com/alexWaligorski/capstone-project/pull/12/files#diff-249192a0ca57c498b7a3d0d783ff5c4f370e11f160d29fd94ebe89ecbe8c875fR12) to call `onSubmit` with the data instead of the default submit event:

```jsx
<StyledForm
  onSubmit={() => {
    const formData = new FormData(event.target);
    const data = Object.fromEntries(formData);
    onSubmit(data);
  }}
>
```

Of course you'd then need to get rid of [the code that does the same thing one level higher in `EditMeetingPage`](https://github.com/alexWaligorski/capstone-project/pull/12/files#diff-775ca6f4fda6fd5f09d3e56acabd366214e519fbfd69ad8103fbfa001e0b8740R23-R24).

Hope this helps. :)

**Reply by alexWaligorski** [reply]
🙏  Wow, thank you for this fast and helpful answer! As much as I hate testing, I really understand the importance of it.

Thanks to your help, my problem has been solved!

I had to add:
`event.preventDefault();`
to your code.

❗And for other students, who want to check if a user can alter entries, which means that the form is prefilled with default data:
don't forget to first clear the input before simulating the user typing in new data:

`  await userEvent.clear(input);
  await user.type(input, "new input");`
